src/roster/views.py

```diff
@@ -1466,3 +1467,64 @@ def remove_social_account(request, *args, **kwargs):
         logger.debug('User %s does not have a %s account', request.user.username, social_app_name)
 
     return HttpResponseRedirect(reverse('my_account'))
+
+class StudentDetailsView(LoginRequiredMixin, ThemedPageMixin, SettingsPageMixin, TemplateView):
+    template_name='roster/student-details.html'
+
+    def __init__(self):
+        super().__init__()
+
+    def get(self, request, *args, **kwargs):
+        self.clusive_user = request.clusive_user
+
+        if not self.clusive_user.can_manage_periods:
+            self.handle_no_permission()
+
+        if 'days' in kwargs:
+            self.days = kwargs.get('days')
+            logger.debug('Setting student activity days = %d', self.days)
+            self.clusive_user.student_activity_days = self.days
+            self.clusive_user.save()
+        else:
+            self.days = self.clusive_user.student_activity_days
+
+        period = self.clusive_user.current_period
+        self.roster = period.users.exclude(user=request.user, role=Roles.TEACHER).order_by('user__first_name')
+        # Dictionary for the individual panel data
+        self.panel_data = dict()
+        self.panel_data['days'] = self.days
+        try:
+            self.clusive_student = ClusiveUser.objects.get(
+                user__username=kwargs['username'],
+                periods__in=[period],
+                role__in=[Roles.STUDENT, Roles.GUEST]
```

Comment:
Guest not needed - would never be a member of a period.

Reply:
Makes sense.

Reply:
Note to self, change to:
```
self.clusive_student = ClusiveUser.objects.get(
    user__username=kwargs['username'],
    periods__in=[period],
    role=Roles.STUDENT
)
```